kafka-coroutines-retry/src/main/kotlin/io/provenance/kafka/coroutines/retry/KafkaErrorHandler.kt

```diff
@@ -0,0 +1,90 @@
+package io.provenance.kafka.coroutines.retry
+
+import io.provenance.kafka.coroutine.UnAckedConsumerRecord
+import io.provenance.kafka.coroutines.retry.store.RetryRecord
+import io.provenance.kafka.coroutines.retry.store.RetryRecordStore
+import io.provenance.kafka.coroutines.retry.store.setHeader
+import java.time.OffsetDateTime
+import kotlinx.coroutines.flow.asFlow
+import mu.KotlinLogging
+import org.apache.kafka.clients.consumer.ConsumerRecord
+
+const val DEFAULT_RECORD_REPROCESS_GROUP_SIZE = 10
+const val KAFKA_RETRY_ATTEMPTS_HEADER = "kcache-retry"
+
+/**
+ * Lift the [FlowProcessor] from [UnAckedConsumerRecord] to [ConsumerRecord] type.
+ *
+ * This is needed to match the flow type of [tryOnEach] when processing with [KafkaFlowRetry].
+ */
+fun <K, V> FlowProcessor<ConsumerRecord<K, V>>.lifted(): FlowProcessor<UnAckedConsumerRecord<K, V>> {
+    return object : FlowProcessor<UnAckedConsumerRecord<K, V>> {
+        override suspend fun send(item: UnAckedConsumerRecord<K, V>) {
+            this@lifted.send(item.toConsumerRecord())
+        }
+
+        override suspend fun process(item: UnAckedConsumerRecord<K, V>, attempt: Int) {
+            this@lifted.process(item.toConsumerRecord())
+        }
+    }
+}
+
+/**
+ * Lift the lambda callback from [UnAckedConsumerRecord] to [ConsumerRecord] type.
+ *
+ * This is needed to match the flow type of [tryOnEach] when processing with [KafkaFlowRetry].
+ */
+fun <K, V> (suspend (ConsumerRecord<K, V>) -> Unit).lifted(): suspend (UnAckedConsumerRecord<K, V>) -> Unit =
```

Comment:
maybe a dumb question, but when you convert the UnAckedConsumerRecord to a ConsumerRecord, what happens with the ack?

Reply:
Nothing, it is maintained in the underlying flow (allowing the `acking()` flow call after the fact). the conversion happens in a `tryOnEach()` (which mirrors the standard kotlin flow's `onEach` method.

Reply:
got it, thanks for the info!